src/descriptor/key.rs

```diff
@@ -485,6 +485,30 @@ impl DescriptorPublicKey {
         DefiniteDescriptorKey::new(definite)
             .expect("The key should not contain any wildcards at this point")
     }
+
+    /// Whether this key matches a [`DefiniteDescriptorKey`]
```

Comment:
In e16e7dd9:

This `matches` function should be private and have a short wrapper
```
pub fn matches<D: AsRef<DescriptorPublicKey>>(&self, key: D) {
    self.real_matches(key.as_ref())
}
```
And then in your `matches` method (now `real_matches`) take a `&DescriptorPublicKey` directly rather than writing `&definite_key.0`.

Reply:
This depends on https://github.com/rust-bitcoin/rust-miniscript/pull/492 BTW

Reply:
I also think that ths `matches` name should be different because I continue to think that this "truncate the path and check for a match" logic is confusing. Maybe it should be called `matches_prefix` or something?

Reply:
I'd be willing to move this to a followup PR because I think this is dominating discussion here, and this PR is actually much much larger than this method :). I'm also musing about whether `DefiniteDescriptorKey` should have say, an `Option<usize>` which indicates which index it came from (if any) to assist with what you're trying to do.

Reply:
The new name is better but I'd still like to revisit this :) BTW I think the code should also have a check that the final (truncated) step in the derivation path is actually a wildcard.